src/vs_seeker.c

```diff
@@ -0,0 +1,776 @@
+    MOVEMENT_ACTION_EMOTE_EXCLAMATION_MARK,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sMovementScript_TrainerNoRematch[] = {
+    MOVEMENT_ACTION_EMOTE_X,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sMovementScript_TrainerRematch[] = {
+    MOVEMENT_ACTION_WALK_IN_PLACE_FASTER_DOWN,
+    MOVEMENT_ACTION_EMOTE_DOUBLE_EXCL_MARK,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sFaceDirectionMovementTypeByFacingDirection[] = {
+    MOVEMENT_TYPE_FACE_DOWN,
+    MOVEMENT_TYPE_FACE_DOWN,
+    MOVEMENT_TYPE_FACE_UP,
+    MOVEMENT_TYPE_FACE_LEFT,
+    MOVEMENT_TYPE_FACE_RIGHT
+};
+
+void VsSeekerFreezeObjectsAfterChargeComplete(void)
+{
+    CreateTask(Task_ResetObjectsRematchWantedState, 80);
+}
+
+static void Task_ResetObjectsRematchWantedState(u8 taskId)
+{
+    struct Task *task = &gTasks[taskId];
+    u8 i;
+
+    if (task->data[0] == 0 && IsPlayerStandingStill() == TRUE)
+    {
+        PlayerFreeze();
+        task->data[0] = 1;
+    }
+
+    if (task->data[1] == 0)
+    {
+        for (i = 0; i < OBJECT_EVENTS_COUNT; i++)
+        {
+            if (ObjectEventIdIsSane(i) == TRUE)
+            {
+                if (gObjectEvents[i].singleMovementActive)
+                    return;
+                FreezeObjectEvent(&gObjectEvents[i]);
+            }
+        }
+    }
+
+    task->data[1] = 1;
+    if (task->data[0] != 0)
+    {
+        DestroyTask(taskId);
+        StopPlayerAvatar();
+        ScriptContext_Enable();
+    }
+}
+
+u16 VsSeekerConvertLocalIdToTableId(u16 localId)
+{
+    u32 localIdIndex = 0;
+    u32 trainerId = 0;
+
+    for (localIdIndex = 0; localIdIndex < OBJECT_EVENTS_COUNT ; localIdIndex++)
+    {
+        if (sVsSeeker->trainerInfo[localIdIndex].localId == localId)
+        {
+            trainerId = sVsSeeker->trainerInfo[localIdIndex].trainerIdx;
+            return TrainerIdToRematchTableId(gRematchTable,trainerId);
+        }
+    }
+    return -1;
+}
+
+void VsSeekerResetObjectMovementAfterChargeComplete(void)
+{
+    struct ObjectEventTemplate * templates = gSaveBlock1Ptr->objectEventTemplates;
+    u8 i;
+    u8 movementType;
+    u8 objEventId;
+    struct ObjectEvent * objectEvent;
+
+    for (i = 0; i < gMapHeader.events->objectEventCount; i++)
+    {
+        if ((templates[i].trainerType == TRAINER_TYPE_NORMAL
+          || templates[i].trainerType == TRAINER_TYPE_BURIED)
+          && (templates[i].movementType == MOVEMENT_TYPE_ROTATE_CLOCKWISE))
+        {
+            movementType = GetRandomFaceDirectionMovementType();
+            TryGetObjectEventIdByLocalIdAndMap(templates[i].localId, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, &objEventId);
+            objectEvent = &gObjectEvents[objEventId];
+            if (ObjectEventIdIsSane(objEventId) == TRUE)
+            {
+                SetTrainerMovementType(objectEvent, movementType);
+            }
+            templates[i].movementType = movementType;
+        }
```

Comment:
this is another example where extracting the (inverted) condition and an early continue would make sense imo